src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
The `MachineProducer` interface should be parameterized to prevent implementations that do not use `Machine` or its subtype. Refer to the checklist: Parameterize `MachineProducer` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created.

Reply:
The MachineProducer interface should be parameterized to prevent implementations that do not use Machine or its subtype. Refer to the checklist: Parameterize MachineProducer in such a way that any implementation NOT using Machine or its subtype could NOT be created.